packages/preferences/src/browser/preferences-tree-widget.ts

```diff
@@ -313,12 +317,19 @@ export class PreferencesEditorsContainer extends DockPanel {
 
         super.onAfterAttach(msg);
         this.onInitEmitter.fire(undefined);
+        this.toDispose.push(
+            this.labelProvider.onDidChange(() => {
+                // Explicitly update the icon for the user preference editor widget.
+                this.userPreferenceEditorWidget.title.iconClass = this.labelProvider.getIcon(new URI('settings.json'));
```

Comment:
what about workspace preference editor widget?

Reply:
> what about workspace preference editor widget?

I originally asked https://github.com/eclipse-theia/theia/pull/7011#issuecomment-580057029 because I was confused as why the workspace icon was updated and the user one wasn't. At the moment the workspace icon does get updated but I can be explicit here as well.